app/models/katello/content_view_erratum_filter.rb

```diff
@@ -94,7 +95,14 @@ def erratum_arel
     def types_clause
       types = erratum_rules.first.types
       return if types.blank?
-      errata_types_in(types)
+
+      conditions = []
+      valid_types = Erratum::TYPES.reject { |type| type == "other" }
+      conditions << errata_types_in(types.reject { |type| type == "other" })
+      conditions << errata_types_not_in(valid_types) if types.include?("other")
+      conditions.reduce(nil) do |combined_clause, condition|
+        combined_clause ? combined_clause.or(condition) : condition
+      end
```

Comment:
I think I understand what this is doing but it's pretty hairy. How can we make it clearer? My issues with it are

1. `Erratum::TYPES.reject { |type| type == "other" }` is too similar to `types.reject { |type| type == "other" }` and that's confusing. Took me a minute to realize that the former is a slice of the constant, and the latter is a slice of the (user?) input. 
2. The single-line `if types.include?("other")` would be more readable as a traditional `if` statement. That way you may even include a comment explaining what this is doing.
3. Can we please get some comments explaining what each line is doing?

This will require a bit of thought. I tried to whip up a quick solution to post as a suggestion here, but one wasn't immediately obvious so this comment is just complaints now, sorry!